src/Compilers/VisualBasic/Portable/Symbols/Source/SourceComplexParameterSymbol.vb

```diff
@@ -154,7 +155,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
 
             Dim defaultSyntax = parameterSyntax.[Default]
             If defaultSyntax Is Nothing Then
-                Return Nothing
+                If InternalSyntax.Feature.ImplicitDefaultValueOnOptionalParameter.IsUnavailable Then
+                    Return Nothing
```

Comment:
This function looks pretty important. I would expect to see code to ensure that at the end the ConstantValue bound is equivalent to binding the literal Nothing, not as if there was no default. The code after this If dots into defaultSyntax assuming it's not null but it can be. Also, in some places the syntax is just used as a place for reporting diagnostics so you should probably pass something for the defaultSyntax to methods that expect something, such as the parameter name (since the As clause won't always be there).